project/ValidatePullRequest.scala

```diff
@@ -3,19 +3,171 @@
  */
 package akka
 
-import sbt._
-import sbt.Keys._
 import com.typesafe.tools.mima.plugin.MimaKeys.reportBinaryIssues
+import net.virtualvoid.sbt.graph.IvyGraphMLDependencies
+import net.virtualvoid.sbt.graph.IvyGraphMLDependencies.ModuleId
+import org.kohsuke.github._
+import sbt.Keys._
+import sbt._
+
+import scala.collection.immutable
+import scala.util.matching.Regex
 
 object ValidatePullRequest extends AutoPlugin {
 
+  val ValidatePR = config("pr-validation") extend Test
```

Comment:
For `test in ValidatePR` to work more is needed than this. `... extend Test` only specifies where to look for settings if a setting is not found under `ValidatePR` scope.

Add all test related tasks under the new configuration scope by:
```
override lazy val projectSettings = inConfig(ValidatePR)(Defaults.testTasks) ++ ...
```

Then add all the settings of `ValidatePR` to every project that has this autoplugin enabled:
```
override lazy val projectConfigurations = Seq(ValidatePR)
```

And finaly, since we have this neat sample build transformation, which unfortunately does not consider AutoPlugins add `.configs(ValidatePullRequest.ValidatePR)` [here](https://github.com/akka/akka/blob/e52ce3afaf2ecee2e68ec1677ec1e29568d320a3/project/Sample.scala#L68).